Network/DNS/API/Resolv.hs

```diff
@@ -0,0 +1,152 @@
+-- Copyright (c) 2013-2015 PivotCloud, Inc.
+--
+-- Network.DNS.API.Resolv
+--
+-- Please feel free to contact us at licensing@pivotmail.com with any
+-- contributions, additions, or other feedback; we would love to hear from
+-- you.
+--
+-- Licensed under the Apache License, Version 2.0 (the "License"); you may
+-- not use this file except in compliance with the License. You may obtain a
+-- copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+--
+-- Unless required by applicable law or agreed to in writing, software
+-- distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+-- WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+-- License for the specific language governing permissions and limitations
+-- under the License.
+--
+
+module Network.DNS.API.Resolv
+    ( getFirstResolvSeed
+      -- * DNS resolver
+    , contactDNSResolver
+    , contactDNSResolverAt
+    , makeResolvSeedSafe
+    ) where
+
+import           Control.Applicative
+import           Control.Exception
+import           Control.Monad.Except
+import           Data.Byteable
+import           Data.ByteString (ByteString)
+import qualified Data.ByteString.Char8 as BC
+import           Data.Hourglass.Types
+import qualified Network.DNS as DNS
+import           Network.DNS.API.Error
+import           Network.DNS.API.FQDN
+import           Network.Socket (PortNumber, HostName)
+
+-- catch all exception
+catchAny :: IO a -> (SomeException -> DnsIO a) -> DnsIO a
```

Comment:
I already looked at this, but the `MonadBaseControl IO m` just stopped me to use it.
The function is so small and the needs so easy to satisfy I rather liked to implement this tiny helper instead of adding more dependancies.